File: src/main/java/com/dashboard/interfaces/serviceInterfaces/EmployeeServiceInterface.java
package com.dashboard.interfaces.serviceInterfaces;

import com.dashboard.models.JPA.Employee;

import java.util.List;
import java.util.Optional;

public interface EmployeeServiceInterface {
    void deleteAllEmployees();

    void addAllEmployees(List<Employee> users);

    Optional<Employee> getByUsername(String username);

    void assignTask(Integer employeeId, Integer taskID);

    void unassignTask(Integer employeeId, Integer taskID);

    long count();
}

File: src/main/kotlin/dashboard/services/interfaces/EmployeeServiceInterface.kt
package dashboard.services.interfaces

import dashboard.models.Employee
import java.util.*

interface EmployeeServiceInterface {
    fun deleteAllEmployees()
    fun addAllEmployees(users: List<Employee>)
    fun addNewEmployee(employee: Employee)
    fun getEmployeeById(id: Int): Optional<Employee>
    fun getAllEmployees(): List<Employee>
    fun totalEmployees(): Long
    fun getByUsername(username: String): Optional<Employee>
    fun deleteEmployeeById(id: Int)
    fun updateEmployee(employeeId: Int, employee: Employee)
    fun assignTask(employeeId: Int, taskID: Int)
    fun unassignTask(employeeId: Int, taskID: Int)
}
